File: electron/handlers/launcher.js
const configService = require('../services/configService');
const claudeCli = require('../services/claudeCli');

function registerLauncherHandlers(ipcMain) {
  ipcMain.handle('launch-claude', async (_event, folder, sessionId) => {
    try {
      const config = configService.loadConfig();

      // Validate that the folder is in the configured list
      if (!config.folders.some((f) => f === folder)) {
        throw new Error('Folder not in configured list');
      }

      const cliCommand = config.claude_cli_command || 'claude';

      if (sessionId) {
        claudeCli.launchResumeSession(folder, sessionId, cliCommand);
      } else {
        claudeCli.launchNewSession(folder, cliCommand);
      }

      return { success: true, folder, sessionId: sessionId || null };
    } catch (err) {
      throw new Error(`Failed to launch Claude: ${err.message}`);
    }
  });
}

module.exports = { registerLauncherHandlers };

File: src-tauri/src/commands/launcher.rs
use crate::errors::AppError;
use crate::services::{claude_cli, config_service};
use std::path::Path;

/// Launch Claude Code (new session or resume existing)
#[tauri::command]
pub async fn launch_claude(
    folder_path: String,
    session_id: Option<String>,
) -> Result<(), String> {
    let config = config_service::load_config().map_err(|e| e.to_string())?;
    if !config.folders.iter().any(|f| f == &folder_path) {
        return Err("Folder not in configured list".to_string());
    }

    let folder = Path::new(&folder_path);
    if !folder.is_dir() {
        return Err(
            AppError::not_found(format!("Folder does not exist: {}", folder_path))
                .to_string(),
        );
    }
    let cli = &config.claude_cli_command;

    match session_id {
        Some(id) => claude_cli::launch_resume_session(cli, folder, &id),
        None => claude_cli::launch_new_session(cli, folder),
    }
    .map_err(|e| e.to_string())
}
